hack/generator/pkg/astmodel/property_definition.go

```diff
@@ -79,41 +79,99 @@ func (property *PropertyDefinition) HasName(name PropertyName) bool {
 
 // WithValidation adds the given validation to the property's set of validations
 func (property *PropertyDefinition) WithValidation(validation Validation) *PropertyDefinition {
-	result := *property
+	result := property.copy()
 	result.validations = append(result.validations, validation)
-	return &result
+	return result
 }
 
 // WithoutValidation removes all validations from the field
 func (property *PropertyDefinition) WithoutValidation() *PropertyDefinition {
-	result := *property
+	result := property.copy()
 	result.validations = nil
-	return &result
+	return result
 }
 
 // WithTag adds the given tag to the field
 func (property *PropertyDefinition) WithTag(key string, value string) *PropertyDefinition {
-	result := *property
-	// TODO: Should we have a copy function here to make this a bit safer? Right now both this function
-	// TODO: and the above WithValidations technically leave references (i.e. maps) the same. That is ok
-	// TODO: as long as this object really is immutable and nothing is changing the content of the maps
-	// TODO: after a copy has been made it it was a bit surprising that we aren't doing a deep copy here.
-	// Have to copy the map here
-	result.tags = make(map[string][]string)
-	for k, v := range property.tags {
-		result.tags[k] = v
+	// Check if exists
+	for _, v := range property.tags[key] {
+		if v == value {
+			return property
+		}
 	}
+
+	result := property.copy()
 	result.tags[key] = append(result.tags[key], value)
```

Comment:
Not 100% sure what you're getting at here:
1. The slice doesn't have the `value` in question (because if it did, the check above would've already returned).
2. The slice can have more than one value, so we append the new one. This is how you append something like `omitempty` when you already have `{"json": []{"foo"}}`

Reply:
I didn't realize the need for multiple values, in order. Makes sense now.